chain/sub/incoming.go

```diff
@@ -203,6 +203,31 @@ func (bv *BlockValidator) Validate(ctx context.Context, pid peer.ID, msg *pubsub
 		}
 	}
 
+	// we check that the miner met the minimum power at the lookback tipset
+
+	baseTs, err := bv.stmgr.ChainStore().LoadTipSet(types.NewTipSetKey(blk.Header.Parents...))
```

Comment:
What if we don't have parents synced (e.g. got a small fork)?

Reply:
We derive this information from our heaviest tipset,

https://github.com/filecoin-project/lotus/blob/c0f0e2ba45cfd36614ba598641b2cf2ea16fbfac/chain/sub/incoming.go#L297-L298

We probably should include this validation as part of `getMinerWorkerKey` and do it all there (and if it fails we can check if `isChainNearSynced` before rejecting).